src/stock-receive-mobile/add-lot-code.jsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+import React, { useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Form } from 'react-final-form';
+import arrayMutators from 'final-form-arrays';
+import { FieldArray } from 'react-final-form-arrays';
+import update from 'immutability-helper';
+
+import BaseField from '../react-components/form-fields/base-field';
+import DateInput from '../stock-adjustment-mobile/components/date-input.component';
+import Input from '../react-components/inputs/input';
+import AddButton from '../react-components/buttons/add-button';
+
+
+const AddLotCodePage = ({ orderableGroupService, lotResource }) => {
+    const history = useHistory();
+    const location = useLocation();
+
+    const menu = document.getElementsByClassName("header ng-scope")[0];
+    menu.style.display = "none";
+
+    useEffect(() => {
+        location.state = location?.state ?? JSON.parse(localStorage.getItem('stateLocationLotCode'));
+    }, [location]);
+
+    const validate = values => {
+        const errors = {};
+
+        if (!values.lotCode) {
+            errors['lotCode'] = { lotCode: 'Required' };
+        }
+
+        if (!values.expirationDate) {
+            errors['expirationDate'] = { expirationDate: 'Required' };
+        }
+
+        return errors;
+    };
+
+    const cancel = () => {
+        const currentFormValues = location.state.currentFormValues.hasOwnProperty('lot') ? update(location.state.currentFormValues, {lot: {['isFromAddLotCodePage']: {$set: true}}}) : location.state.currentFormValues;
```

Comment:
I would break it into multiple lines:
```
const currentFormValues = location.state.currentFormValues.hasOwnProperty('lot')
  ? update(location.state.currentFormValues, {lot: {['isFromAddLotCodePage']: {$set: true}}})
  : location.state.currentFormValues;
```